acvm-repo/brillig_vm/src/lib.rs

```diff
@@ -384,14 +394,18 @@ impl<'a, F: AcirField, B: BlackBoxFunctionSolver<F>> VM<'a, F, B> {
             Opcode::ConditionalMov { destination, source_a, source_b, condition } => {
                 let condition_value = self.memory.read(*condition);
 
-                let condition_value_bool =
-                    condition_value.expect_u1().expect("condition value is not a boolean");
-                if condition_value_bool {
+                let condition_value = match condition_value.expect_u1() {
+                    Err(error) => {
+                        return self.fail(format!("condition value is not a boolean: {error}"));
+                    }
+                    Ok(cond) => cond,
+                };
+                if condition_value {
```

Comment:
For consistency with the above case, maybe this could also match against `Ok(true)` and `Ok(false)` (a minor thing, really)